sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionHelper.scala

```diff
@@ -400,6 +401,9 @@ abstract class TypeCoercionHelper {
         NaNvl(Cast(l, DoubleType), r)
       case NaNvl(l, r) if r.dataType == NullType => NaNvl(l, Cast(r, l.dataType))
 
+      case r: RandStr if r.length.dataType != IntegerType =>
+        r.copy(length = Cast(r.length, IntegerType))
```

Comment:
not all the implicit cast should be added, let's reuse the existing util functions:
```
case r: RandStr if r.length.dataType != IntegerType =>
  implicitCast(r.length, IntegerType).map { casted =>
    r.copy(length = casted)
  }.getOrElse(r)
```

Reply:
sounds good, done.